src/main/java/org/jvnet/hudson/test/PluginAutomaticTestBuilder.java

```diff
@@ -106,5 +126,88 @@ public void testPluginActive() {
             }
         }
 
+        public void testStaplerDispatches() throws InvocationTargetException, IllegalAccessException, NoSuchMethodException {
+            List<String> methodsFound = new ArrayList<>();
+
+            Method isStaplerRoutableMethod = findIsRoutableMethod();
+            if (isStaplerRoutableMethod == null) {
+                return;
+            }
+
+            PluginWrapper thisPlugin = determineCurrentPlugin();
+            if (thisPlugin == null) {
+                return; // we're not in a plugin
+            }
+            ClassGraph classGraph = new ClassGraph();
+            classGraph.addClassLoader(thisPlugin.classLoader);
+            classGraph.disableJarScanning();
+            classGraph.enableClassInfo();
+            ScanResult result = classGraph.scan();
+
+            for (ClassInfo classInfo : result.getAllClasses()) {
+                Class clazz = classInfo.loadClass();
+                for (Method m : clazz.getDeclaredMethods()) {
+                    if (isStaplerDispatchable(m) && (boolean)isStaplerRoutableMethod.invoke(null, m)) {
+                        if (!hasStaplerVerbAnnotation(m)) {
+                            methodsFound.add(clazz.getName() + "#" + m.getName());
+                        }
+                    }
+                }
+            }
+            Assert.assertThat("There should be no web methods that lack HTTP verb annotations like @RequirePOST, @GET, @POST, etc. -- see https://jenkins.io/redirect/developer/csrf-protection", methodsFound, is(empty()));
+        }
+
+        private Method findIsRoutableMethod() throws NoSuchMethodException {
+            try {
+                Method method = Class.forName("jenkins.security.stapler.TypedFilter").getDeclaredMethod("isRoutableMethod", Method.class);
+                method.setAccessible(true);
+                return method;
+            } catch (ClassNotFoundException e) {
+                LOGGER.warning("This test requires Jenkins 2.154, Jenkins LTS 2.138.4, or newer to run, use e.g. -Djenkins.version=2.138.4");
```

Comment:
Use of `assume` from JUnit might be appropriate here for better integration with error reporting.